code/go/0chain.net/smartcontract/minersc/globals.go

```diff
@@ -0,0 +1,450 @@
+	GlobalSettingName[AsyncFetchingMaxSimultaneousFromSharders]: {smartcontract.Int, false},
+
+	GlobalSettingName[HealthCheckDeepScanEnabled]:                 {smartcontract.Boolean, false},
+	GlobalSettingName[HealthCheckDeepScanBatchSize]:               {smartcontract.Int64, false},
+	GlobalSettingName[HealthCheckDeepScanWindow]:                  {smartcontract.Int64, false},
+	GlobalSettingName[HealthCheckDeepScanSettleSecs]:              {smartcontract.Duration, false},
+	GlobalSettingName[HealthCheckDeepScanIntervalMins]:            {smartcontract.Duration, false},
+	GlobalSettingName[HealthCheckDeepScanReportStatusMins]:        {smartcontract.Duration, false},
+	GlobalSettingName[HealthCheckProximityScanEnabled]:            {smartcontract.Boolean, false},
+	GlobalSettingName[HealthCheckProximityScanBatchSize]:          {smartcontract.Int64, false},
+	GlobalSettingName[HealthCheckProximityScanWindow]:             {smartcontract.Int64, false},
+	GlobalSettingName[HealthCheckProximityScanSettleSecs]:         {smartcontract.Duration, false},
+	GlobalSettingName[HealthCheckProximityScanRepeatIntervalMins]: {smartcontract.Duration, false},
+	GlobalSettingName[HealthCheckProximityScanRejportStatusMins]:  {smartcontract.Duration, false},
+	GlobalSettingName[HealthCheckShowCounters]:                    {smartcontract.Boolean, false},
+}
+
+var GLOBALS_KEY = datastore.Key(encryption.Hash("global_settings"))
+
+func scConfigKey(scKey string) datastore.Key {
+	return datastore.Key(scKey + ":configurations")
+}
+
+type GlobalSettings struct {
+	Fields map[string]string `json:"fields"`
+}
+
+func newGlobalSettings() *GlobalSettings {
+	return &GlobalSettings{
+		Fields: make(map[string]string),
+	}
+}
+
+// Encode implements util.Serializable interface.
+func (gl *GlobalSettings) Encode() []byte {
+	var b, err = json.Marshal(gl)
+	if err != nil {
+		panic(err)
+	}
+	return b
+}
+
+// Decode implements util.Serializable interface.
+func (gl *GlobalSettings) Decode(p []byte) error {
+	return json.Unmarshal(p, gl)
+}
+
+func (gl *GlobalSettings) save(balances cstate.StateContextI) error {
+	_, err := balances.InsertTrieNode(GLOBALS_KEY, gl)
+	return err
+}
+
+func (gl *GlobalSettings) update(inputMap smartcontract.StringMap) error {
+	var err error
+	for key, value := range inputMap.Fields {
+		info, found := GlobalSettingInfo[key]
+		if !found {
+			return fmt.Errorf("'%s' is not a valid global setting", key)
+		}
+		if !info.mutable {
+			return fmt.Errorf("%s is an immutable setting", key)
+		}
+		_, err = smartcontract.StringToInterface(value, info.settingType)
+		if err != nil {
+			return fmt.Errorf("%v value %v cannot be perased as a %s",
+				key, value, smartcontract.ConfigTypeName[info.settingType])
+		}
+		gl.Fields[key] = value
+	}
+	return nil
+}
+
+func (gl *GlobalSettings) GetInt(field GlobalSetting) int {
+	sValue, found := gl.Fields[GlobalSettingName[field]]
+	if !found {
+		return viper.GetInt(GlobalSettingName[field])
```

Comment:
I'm thinking if we get value from viper (local yaml file) is okay here, as when a new miner joined the network, it's highly possible that the miner does not have the full MPT, and could not get the value from MPT. If we get it from local yaml file, which may different from the value in MPT, the miner could behavior differently and cause unexpected behaviors.

Reply:
```
We could have partial state of MPT for new joined miners,  basically we can not sync the whole MPT from remote when it starts, it is highly possible that the miner could not get settings from the MPT, we shall wait till MPT has enough state to work. We may have changed the settings in MPT, if the new joined miner get settings from local, it would use different settings from others, which could lead problems. So the situations are:
a) The chain just started, we should init the MPT with local settings
b) The chain has run for a while, there are changes applied to the settings, a new miner joined the network, before it syncs up the MPT state,  it could not get settings from MPT, however, we shall not use local settings as well, because the settings could be different from others, and could cause problem if it use the different settings to participant the block VRF share verification, and block generation. (
```